addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaBridgeHandler.java

```diff
@@ -216,7 +227,7 @@ void updateAllZoneInformation() {
                 // If thing still thinks that the bridge is offline, update its status.
                 if (thing.getStatusInfo().getStatusDetail() == ThingStatusDetail.BRIDGE_OFFLINE) {
                     handler.bridgeStatusChanged(ThingStatusInfoBuilder.create(bridge.getStatus()).build());
```

Comment:
The NPE never occurred though, due to the `thing.getStatusInfo().getStatusDetail() == ThingStatusDetail.BRIDGE_OFFLINE`. However, since you pointed that out, it might be more elegant to wrap the entire loop body in a check if handler is already assigned, like this:
```
for (Thing thing : bridge.getThings()) {
	YamahaZoneThingHandler handler = (YamahaZoneThingHandler) thing.getHandler();

	// ensure the handler has beed already assigned
	if (handler != null) {
		// If thing still thinks that the bridge is offline, update its status.
		if (thing.getStatusInfo().getStatusDetail() == ThingStatusDetail.BRIDGE_OFFLINE) {                    					 
			handler.bridgeStatusChanged(ThingStatusInfoBuilder.create(bridge.getStatus()).build());
		} else if (handler != null && handler.isCorrectlyInitialized()) {
			handler.updateZoneInformation();
		}
	}
}
```

What do you think?

Reply:
Yes, because the `getHandler()` method has a `@Nullable` annotation which means it could return `null` at any time.

See https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/Thing.java#L132-L138.

Reply:
The change has now been applied.